modules/model/shared/src/main/scala/observe/model/ExecutionState.scala

```diff
@@ -28,7 +26,7 @@ case class ExecutionState(
   observer:        Option[Observer],
   runningStepId:   Option[Step.Id],
   nsState:         Option[NsRunningState],
-  stepResources:   List[(Step.Id, List[(Resource | Instrument, ActionStatus)])],
+  stepResources:   Map[Step.Id, Map[Resource | Instrument, ActionStatus]],
```

Comment:
I found it easier to deal with this if we model as a `Map` (allows for example `index`/`at` optics).

Reply:
👍